### PR TITLE
[pipelineX](refactor) refine relationship between shared state and dependency

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/pipeline/pipeline_x/dependency.h [hunk 1]
```diff
@@ -57,22 +57,22 @@ static constexpr auto TIME_UNIT_DEPENDENCY_LOG = 30 * 1000L * 1000L * 1000L;
...
-
-    std::atomic_bool source_released_flag {false};
-    std::atomic_bool sink_released_flag {false};
-    std::mutex source_release_lock;
-    std::mutex sink_release_lock;
-
-    void release_source_dep() {
-        std::unique_lock<std::mutex> lc(source_release_lock);
-        source_released_flag = true;
+    template <class TARGET>
```
Comment (BiteTheDDDDt) [inline be/src/pipeline/pipeline_x/dependency.h:60]: Target

**Review by zhangstar333 — APPROVED** [review]

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**